src/test/java/org/javarosa/core/model/DateTimeTest.java

```diff
@@ -0,0 +1,121 @@
+package org.javarosa.core.model;
+
+import org.javarosa.core.model.data.DateData;
+import org.javarosa.core.model.data.DateTimeData;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.data.TimeData;
+import org.javarosa.test.Scenario;
+import org.javarosa.xform.parse.XFormParser;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.javarosa.test.BindBuilderXFormsElement.bind;
+import static org.javarosa.test.XFormsElement.body;
+import static org.javarosa.test.XFormsElement.head;
+import static org.javarosa.test.XFormsElement.html;
+import static org.javarosa.test.XFormsElement.input;
+import static org.javarosa.test.XFormsElement.mainInstance;
+import static org.javarosa.test.XFormsElement.model;
+import static org.javarosa.test.XFormsElement.t;
+import static org.javarosa.test.XFormsElement.title;
+
+public class DateTimeTest {
+    @Test
+    public void timeQuestionReturnsTimeDataAnswer() throws IOException, XFormParser.ParseException {
+        Scenario scenario = Scenario.init("DateTime form", html(
+            head(
+                title("Time form"),
+                model(
+                    mainInstance(t("data id=\"time-form\"",
+                        t("time1"),
+                        t("time2", "23:14:00.000+02:00"),
+                        t("time3")
+                    )),
+                    bind("/data/time1").type("time").calculate("&quot;23:14:00.000+02:00&quot;"),
+                    bind("/data/time2").type("time"),
+                    bind("/data/time3").type("time").calculate("/data/time2")
+                )
+            ),
+            body(
+                input("/data/time1"),
```

Comment:
Just to make these tests a little more readable, maybe we could name the questions around what they test. Instead of `time1` and so on, it could be:

- `calculateLiteral`
- `empty`
- `calculateReference`

Reply:
Done.